src/audio/module_adapter/module/iadk_modules.c

```diff
@@ -80,8 +83,33 @@ static int iadk_modules_init(struct processing_module *mod)
 	uint32_t instance_id = IPC4_INST_ID(mod->dev->ipc_config.id);
 	uint32_t log_handle = (uint32_t) mod->dev->drv->tctx;
 	/* Connect loadable module interfaces with module adapter entity. */
-	void *mod_adp = system_agent_start(md->module_entry_point, module_id,
-					   instance_id, 0, log_handle, &mod_cfg);
+	/* Check if native Zephyr lib is loaded */
+	struct sof_man_fw_desc *desc;
+
+	desc = lib_manager_get_library_module_desc(module_id);
+	if (!desc) {
+		comp_err(dev, "iadk_modules_init(): Failed to load manifest");
+		return -ENOMEM;
+	}
+	struct sof_man_module *module_entry =
```

Comment:
Do we validate entry is in the module memery region ?

Reply:
We can, but it is not part of that PR.